packages/babel-register/src/node.js

```diff
@@ -41,7 +41,7 @@ function compile(code, filename) {
   const opts = new OptionManager().init(
     // sourceRoot can be overwritten
     {
-      sourceRoot: path.dirname(filename),
+      sourceRoot: path.dirname(filename) + path.sep,
```

Comment:
The `sourceRoot` should be a valid URL. (C.f. https://github.com/mozilla/source-map/issues/340#issuecomment-392879514) Ideally we should use `url.filePathToURL` shipped in node.js 10.12 or [file-url](https://github.com/sindresorhus/file-url) to convert `path.dirname` into file URL and then add a trailing slash, because `\` is not valid in URL.

Sidenote: Although the spec says `sourceRoot` is supposed to prepend sources, the `source-map` library support sourceRoot without slash: https://github.com/mozilla/source-map/blob/7a263408de1b60e60b6220ca3e36d9b3e4aa047d/lib/util.js#L413

Reply:
Babel is currently outputting `C:\Users\...` for `sourceRoot` on windows, and `path.sep` is consistent with that.

Changing `sourceRoot` to be a file URL isn’t something that I have time to code and test right now.

I hope that won’t block the merging of this bug fix.